src/logs/self_monitoring/sfm.py

```diff
@@ -102,12 +102,23 @@ def _generate_metrics(self):
             "Value": self._function_name,
         }]
 
-        metrics.append(_prepare_cloudwatch_metric(
-            "Kinesis record age", self._kinesis_records_age, "Seconds", common_dimensions))
-        metrics.append(_prepare_cloudwatch_metric(
-            "Kinesis record.data compressed size", self._record_data_compressed_size, "Bytes", common_dimensions))
-        metrics.append(_prepare_cloudwatch_metric(
-            "Kinesis record.data decompressed size", self._record_data_decompressed_size, "Bytes", common_dimensions))
+        metrics.append(_prepare_cloudwatch_metric_statistic(
+            "Kinesis record age", "Seconds", common_dimensions,
+            min(self._kinesis_records_age), max(self._kinesis_records_age),
+            sum(self._kinesis_records_age), len(self._kinesis_records_age)
```

Comment:
min, max, sum, len would fit better inside the method imho, with just the proper metric (self._kinesis_records_age) being passed

Reply:
Great idea!

Reply:
You are right, much cleaner, fixed